File: todo-backend/src/main/java/com/todoManagement/todo/service/TodoService.java
package com.todoManagement.todo.service;


import java.util.List;

import com.todoManagement.todo.dto.TodoDto;
import com.todoManagement.todo.entity.Todo;


public interface TodoService {

    TodoDto addTodo(TodoDto todo);

    TodoDto getTodo(Long id);

    List<TodoDto> getAllTodos();

    TodoDto updateTodo(Long id, TodoDto todo);

    void deleteTodo(Long id);

    TodoDto completeTodo(Long id);

    TodoDto inCompleteTodo(Long id);
}

File: todo_management/todo_frontend/todo-ui/src/services/TodoService.js
import axios from "axios";
import { getToken } from "./AuthService";

const REST_API_BASE_URL = 'http://localhost:8080/api/todos';


axios.interceptors.request.use(function (config) {

    config.headers['Authorization'] = getToken();

    return config;
}, function (error){
    return Promise.reject(error);
});

export const getAllTodos = () => axios.get(REST_API_BASE_URL);

export const createTodo = (todo) => axios.post(REST_API_BASE_URL, todo);

export const getTodo = (id) => axios.get(REST_API_BASE_URL + '/' + id);

export const updateTodo = (id, todo) => axios.put(REST_API_BASE_URL + '/' + id, todo);

export const deleteTodo =(id) => axios.delete(REST_API_BASE_URL + '/' + id);

export const completeTodo = (id) => axios.patch(REST_API_BASE_URL + '/' + id + '/complete');

export const InCompleteTodo = (id) => axios.patch(REST_API_BASE_URL + '/' + id + '/in-complete');
